diff --git a/src/hb-ot-shape.cc b/src/hb-ot-shape.cc
index c4027c4a7..9d76e1141 100644
--- a/src/hb-ot-shape.cc
+++ b/src/hb-ot-shape.cc
@@ -179,12 +179,14 @@ hb_ot_shape_planner_t::compile (hb_ot_shape_plan_t           &plan,
 #endif
 #ifndef HB_NO_OT_KERN
     else if (hb_ot_layout_has_kerning (face))
-      plan.apply_kern = script_fallback_position; // Not all shapers apply legacy `kern`
+      plan.apply_fallback_kern = true;
 #endif
     else {}
   }
 
-  plan.apply_fallback_kern = script_fallback_position && !(plan.apply_gpos || plan.apply_kerx || plan.apply_kern);
+  plan.apply_fallback_kern = plan.apply_fallback_kern ||
+                             (script_fallback_position &&
+                              !(plan.apply_gpos || plan.apply_kerx || plan.apply_kern));
 
   plan.zero_marks = script_zero_marks &&
 		    !plan.apply_kerx &&
